src/ident/AclIdent.cc

```diff
@@ -56,10 +54,8 @@ int
 ACLIdent::match(ACLChecklist *cl)
 {
     const auto checklist = Filled(cl);
-    if (checklist->rfc931[0]) {
-        return data->match(checklist->rfc931);
-    } else if (checklist->conn() != nullptr && checklist->conn()->clientConnection != nullptr && checklist->conn()->clientConnection->rfc931[0]) {
-        return data->match(checklist->conn()->clientConnection->rfc931);
+    if (const auto ident = checklist->ident()) {
+        return data->match(ident.value().isEmpty() ? dash_str : SBuf(ident.value()).c_str());
```

Comment:
The data here is ACLUserData - so the 'REQUIRED' special value will be handled as it is documented -we should not remove it (my tests confirmed that).

Reply:
Also I [undone](https://github.com/measurement-factory/squid/pull/245/commits/41492061fa14a638f3537a5c50a7a7bd6cad23e7) the documentation changes about "-" special meaning. ACLIdent::data is  ACLUserData, and ACLUserData::match() returns false if it gets nil or "-" ident user value before trying to match it with the configured ACL value (i.e, dash). We do not want to change this behavior in this PR, I assume. Note that master Squid cannot not hit this scenario because it aborts before it with our assertion:
```
assertion failed: FilledChecklist.cc:273: "!rfc931[0]"
    connection: conn17 local=127.0.0.1:3128 remote=127.0.0.1:58930 FD 13 flags=1 IDENT::-
```
It can be checked in Squid v5 which does not assert and behaves as the PR in this respect (i.e., does not match dash)